File: src/test/java/net/flyingfishflash/loremlist/unit/domain/LrmComponentTypeTests.java
package net.flyingfishflash.loremlist.unit.domain;

import static org.assertj.core.api.Assertions.assertThat;

import net.flyingfishflash.loremlist.domain.LrmComponentType;
import org.junit.jupiter.api.Test;

class LrmComponentTypeTests {

  @Test
  void inverseOfListIsItem() {
    assertThat(LrmComponentType.List.invert()).isEqualTo(LrmComponentType.Item);
  }

  @Test
  void inverseOfItemIsList() {
    assertThat(LrmComponentType.Item.invert()).isEqualTo(LrmComponentType.List);
  }

  @Test
  void inverseOfListItemIsListItem() {
    assertThat(LrmComponentType.ListItem.invert()).isEqualTo(LrmComponentType.ListItem);
  }
}

File: src/test/kotlin/net/flyingfishflash/loremlist/unit/domain/LrmComponentTypeTests.kt
package net.flyingfishflash.loremlist.unit.domain

import io.kotest.core.spec.style.DescribeSpec
import io.kotest.matchers.shouldBe
import net.flyingfishflash.loremlist.domain.LrmComponentType

class LrmComponentTypeTests : DescribeSpec({

  describe("invert()") {
    it("inverse of Item is List") {
      val inverse = LrmComponentType.List.invert()
      inverse.shouldBe(LrmComponentType.Item)
    }

    it("inverse of List is Item") {
      val inverse = LrmComponentType.Item.invert()
      inverse.shouldBe(LrmComponentType.List)
    }
  }
})
